rllab/envs/mujoco/dynamics_randomization/variation.py

```diff
@@ -0,0 +1,241 @@
+from enum import Enum
+
+
+class Method(Enum):
+    """
+    The random coefficient is applied according to these methods.
+    """
+    """ The randomization is the product of the coefficient and the dynamic parameter """
+    COEFFICIENT = 1
+    """ The randomization is equal to the coefficient """
+    ABSOLUTE = 2
+
+
+class Distribution(Enum):
+    """
+    The different ways to produce the random coefficient.
+    """
+    """ Gaussian distribution """
+    GAUSSIAN = 1
+    """ Uniform distribution """
+    UNIFORM = 2
+
+
+class Variation:
+    """
+    Each dynamic parameter to randomize is represented by a Variation. This
+    class works more like a data structure to store the data fields required
+    to find the dynamic parameter and the randomization to apply to it.
+    """
+
+    def __init__(self,
+                 xpath=None,
+                 attrib=None,
+                 method=None,
+                 distribution=None,
+                 var_range=None,
+                 mean_std=None,
+                 elem=None,
+                 default=None):
```

Comment:
Can we have some defaults here to make it a bit easier to use? Method is probably absolute in most cases and distribution is uniform?

Reply:
Default values were assigned in VariationSpec, which is the version of Variation with setters that implements the fluent interface, specifically for fields method, distribution, mean_std and var_range. Fields xpath, attrib, elem and default are very specific to the model in the XML file provided by the user, so they cannot be default parameters. Further more, elem and default are obtained by parsing the XML file, so the user won't set them.